### DISCUSSION TITLE
How to enable pointer event to trigger when a control renders nothing?

### DISCUSSION THREAD

**Discussion by Supheria** [AvaloniaUI/Avalonia#17855] [Q&A] [answered]
When nothing rendered in Render(), saying no child and `context` does nothing, it seems pointer event such as `PointerPressed` cannot be triggered. No matter using `PointerPressed +=` or `override void OnPointerPressed(PointerPressedEventArgs e)` doesn't work.

A solution I find is to fill a background using `context.FillRectangle`, is there any other way?

> The reason I want to do this is I prefer to use `context.Custom` with `SKCanvas` to render a custom control, however pointer events don't response in such case.

**Comment by maxkatz6** [comment, marked as answer]
You can override HitTest for custom draw operation:
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Base/Rendering/SceneGraph/CustomDrawOperation.cs#L24

**Reply by maxkatz6** [reply]
Rect Bounds { get; } should also return correct value

**Reply by Supheria** [reply]
Thank you. Pointer and gesture events response when set HitTest to return true.
